asm_utils/src/reachability.rs

```diff
@@ -54,7 +56,7 @@ pub fn filter_reachable_from<R: Register, F: FunctionOpKind>(
         .collect();
 }
 
-pub fn find_reachable_labels<'a, R: Register, F: FunctionOpKind>(
+pub fn find_reachable_labels<'a, R: Register, F: FunctionOpKind, A: Architecture>(
```

Comment:
Are `Register`, `FunctionOpKind` and `Architecture` orthogonal? Or could we have something like

```
trait Architecture {
   type Register: Register;
   type FunctionOpKind: FunctionOpKind;
}
```

And reduce the number of generic parameters on the functions in this file.

Not blocking this PR though.